### PR TITLE
feat: support creating read-only OCI store from `fs.FS`

### LINKED ISSUE #336: refactor `content/oci` with `fs.FS`
Refactor the `content/oci` package so that it is based on [io/fs.FS](https://pkg.go.dev/io/fs#FS).

With such refactoring, we can have something like
```go
func NewFromFS(fsys fs.FS) *ReadOnlyStore
func NewStorageFromFS(fsys fs.FS) *ReadOnlyStorage
```

Powered by `fs.FS`, we can further leverage [embed.FS](https://pkg.go.dev/embed#FS) and [zip.Reader](https://pkg.go.dev/archive/zip#Reader) to read OCI layout from various file systems.

If we implement `fs.FS` for tarballs, we can directly read OCI layout from a tarball generated by `docker buildx build --output=type=oci,dest=oci.tar`.

```console
$ tar xvf oci.tar
blobs/
blobs/sha256/
blobs/sha256/4f4fb700ef54461cfa02571ae0db9a0dc1e0cdb5577484a6d75e68dc38e8acc1
blobs/sha256/643c10cea5330d08b25f85e99a1f9158efae4a2262125daf3e815a2fddb97ac7
blobs/sha256/8c356d8c5bb84bb001fc8e6c77474833d2b064d21de8610fa26a3b1da16a7921
blobs/sha256/a9c898c2b650367087223a0e5a12260ab88d309af8461fbfaac4ece4f4fc3c71
blobs/sha256/c2eb24acba5f5d4e6a231bff0b8e97050aa081be4dae2fad620896766617ae0e
blobs/sha256/df9b9388f04ad6279a7410b85cedfdcb2208c0a003da7ab5613af71079148139
blobs/sha256/f4612cfdf7a5b92df210c3ce1f0338b651b148023215f40b5b714f0444a0e823
index.json
manifest.json
oci-layout
```

Reference: [Go 1.16 Release Notes](https://go.dev/doc/go1.16#fs)

### PR DESCRIPTION
**Pull request by Wwwsylvia**
Will support creating OCI store from tarball in next PR.

Resolves: #336 
Signed-off-by: Lixia (Sylvia) Lei <lixlei@microsoft.com>

### CODE REVIEW

File: content/oci/readonlystorage.go [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+	if err := dgst.Validate(); err != nil {
+		return "", fmt.Errorf("cannot calculate blob path from invalid digest %s: %v", dgst.String(), err)
+	}
+
+	// NOTE: DirFS does not support opening paths with Windows separators
+	return strings.Join([]string{
+		"blobs",
+		dgst.Algorithm().String(),
+		dgst.Encoded()},
+		"/"), nil
```
[the remark this answers is not in the document]
Comment (Wwwsylvia) [reply content/oci/readonlystorage.go:98]: Great!

File: content/oci/readonlystorage.go [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+	return true, nil
+}
+
+// blobPath calculates blob path from the given digest.
+func blobPath(dgst digest.Digest) (string, error) {
+	if err := dgst.Validate(); err != nil {
+		return "", fmt.Errorf("cannot calculate blob path from invalid digest %s: %v", dgst.String(), err)
+	}
+	// NOTE: Some FSs (e.g. DirFS) do not support opening paths with Windows separators
```
Comment (shizhMSFT) [inline content/oci/readonlystorage.go:86]: For `fs`, we always use slash-separated paths, even for windows.

Comment (shizhMSFT) [reply content/oci/readonlystorage.go:86]: Since `blobPath` is an internal method, we can remove this comment.

Comment (shizhMSFT) [reply content/oci/readonlystorage.go:86]: Note that on Windows,
```go
package main

import (
	"fmt"
	"path/filepath"
)

func main() {
	path := filepath.Join(`foo\bar`, "a/b/c")
	fmt.Println(path)
}
```
outputs
```
foo\bar\a\b\c
```

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Pull request merged**